lib/cocoapods/resolver.rb

```diff
@@ -336,7 +348,9 @@ def indicate_progress
     # @return [Array<Specification>] List of specifications satisfying given requirements.
     #
     def specifications_for_dependency(dependency, additional_requirements = [])
-      requirement = Requirement.new(dependency.requirement.as_list + additional_requirements.flat_map(&:as_list))
+      requirement_list = dependency.requirement.as_list + additional_requirements.flat_map(&:as_list)
+      requirement_list.uniq!
+      requirement = Requirement.new(requirement_list)
```

Comment:
is it more performant to do it in-place instead of passing `requirements.uniq`?

Reply:
yup, saves on an object allocation, and it's safe to do since `+` returns a new instance